### PR TITLE
[Statistics] Rework code to add checks and fix notices/warnings

### PR DESCRIPTION
**Pull request by johnsaigle**
This module currently throws up notices and warnings everywhere. This PR addresses these.

The module is still fairly inconsistent and much of its functionality could be refactored into functions, for example. However, after this PR the errors and warnings are more descriptive and will assist in further modifications

**This PR**:

- [x] Add checks to prevent notices and warnings

- [x] Fixes typos

- [x] Refactors some bizarre coding choices

- [x] Suggests items for further refactoring

- [x] Doesn't affect current functionality

- [x] Passes phpcs

### CODE REVIEW

**Comment by johnsaigle** [comment]
note: Related to #2813 and will probably create a conflict

File: modules/statistics/php/NDB_Form_statistics.class.inc [hunk 1]
```diff
@@ -99,12 +100,25 @@ class NDB_Form_Statistics extends NDB_Form
...
-            $subproj = $row['SubprojectID'];
-            $vl      = $row['VLabel'];
-            $subcat  = $row['Subcat'];
-            $center  = $row['CenterID'];
+            // TODO: The below fields are not always present
+            // in $data, e.g., in the $result for imaging statistics
+            // This should probably be refactored in the future,
+            // But for now these checks exist to eliminate
+            // PHP Notice noise when the keys are not present
+            if (array_key_exists('SubprojectID', $row)) {
```
Comment (jstirling91) [inline modules/statistics/php/NDB_Form_statistics.class.inc]: this could be done using null coalescing operator

**Comment by ZainVirani** [comment]
code looks good, testing now

**Comment by ZainVirani** [comment]
cant test until its rebased @ridz1208

**Comment by ZainVirani** [comment]
My rebase destroyed this PR, HOWEVER
It was made before Xavier and Dave's Statistics module refactor #3240 which I believe doomed it from the start.
I'm going to work John's changes into the new module in a brand new PR.

**Comment by ridz1208** [comment]
indeed seems like the file being modified here has moved and hopefully has been cleaned
this should be closed

**Pull request closed without merging**